### ISSUE TITLE
Proposal: DockPanel for WinUI 3

### ISSUE THREAD

**Issue by nlogozzo** [microsoft/microsoft-ui-xaml#5188]
It would be nice to see a DockPanel in WinUI 3 similar to WPF and the Windows Community Toolkit. 

And no, I can't use the WCT in WinUI 3 since I'm using C++

**Comment by StephenLPeters** [comment]
@chigy FYI

**Comment by mdtauk** [comment]
I wonder if the TabView Tab tearing code could be used for the docking and undocking of the panel.  There may need to be a grabber, like with the Win32 Common Controls Rebar
![image](https://user-images.githubusercontent.com/7389110/122149293-c35a1400-ce53-11eb-9bf5-8a9480ba2849.png)
![image](https://user-images.githubusercontent.com/7389110/122149340-d1a83000-ce53-11eb-8ea8-95a1786f8994.png)

**Comment by jschroedl** [comment]
Really want this for the same reason as @nlogozzo  -- using C++/WinRT.

**Comment by nlogozzo** [comment]
> Really want this for the same reason as @nlogozzo  -- using C++/WinRT.

Yeah I've since switched to Avalonia and C#. C# cause obviously nothing modern wants to support C++ even though it's the back and of everything lol 🤷🏻‍♂️. And Avalonia because it's cross platform, has the new WinUI styles, and is a lot more performant for desktop compared to WndowsAppSdk

**Comment by heartacker** [comment]
come form wpf ，and we need this

**Comment by hkhangus** [comment]
I come from 2023, and this feature hasn't added yet 😢

**Comment by Tartarusome** [comment]
Coming from 2024,i 'd make this component by myself.
(You can never underestimate Microsoft’s development speed, because it can always go even lower.)

**Comment by ghost1372** [comment]
https://github.com/qian-o/WinUI.Dock
